src/execution/window_executor.cpp

```diff
@@ -7,6 +7,75 @@
 
 namespace duckdb {
 
+//===--------------------------------------------------------------------===//
+// WindowDataChunk
+//===--------------------------------------------------------------------===//
+bool WindowDataChunk::IsSimple(const Vector &v) {
+	switch (v.GetType().InternalType()) {
+	case PhysicalType::BOOL:
+	case PhysicalType::UINT8:
+	case PhysicalType::INT8:
+	case PhysicalType::UINT16:
+	case PhysicalType::INT16:
+	case PhysicalType::UINT32:
+	case PhysicalType::INT32:
+	case PhysicalType::UINT64:
+	case PhysicalType::INT64:
+	case PhysicalType::FLOAT:
+	case PhysicalType::DOUBLE:
+	case PhysicalType::INTERVAL:
+	case PhysicalType::LIST:
```

Comment:
Can lists be handled without locking?

Reply:
I don't think so. Lists, arrays, structs and strings all involve reallocating shared memory data structures, so I assumed they needed locking.  Lists are especially bad because they use offsets into a shared data structure. Strings might be made to work by keeping the allocations in the local states, and we might be able to do something recursive with structs, but lists seemed tricky.

Reply:
Should this then return `false` for lists as well?

Reply:
Oops!  Good catch.Guess we don't have any tests at scale...

I was wondering if there was already a predicate somewhere that answers this?

Reply:
There's `LogicalType::IsNested`